src/main/scala/org/apache/spark/sql/pulsar/PulsarProvider.scala

```diff
@@ -562,18 +589,20 @@ private[pulsar] object PulsarProvider extends Logging {
   }
 
   private def prepareConfForReader(parameters: Map[String, String])
-    : (ju.Map[String, Object], ju.Map[String, Object], String, String) = {
+    : (ju.Map[String, Object], ju.Map[String, Object], ju.Map[String, Object], String, String) = {
 
     val serviceUrl = getServiceUrl(parameters)
     val adminUrl = getAdminUrl(parameters)
 
     var clientParams = getClientParams(parameters)
     clientParams += (SERVICE_URL_OPTION_KEY -> serviceUrl)
     val readerParams = getReaderParams(parameters)
+    val adminParams = if (hasAdminParams(parameters)) getAdminParams(parameters) else clientParams
```

Comment:
I don't think this is a good idea. `hasAdminParams` just call the `getAdminParams` and filter. This would cause the real getAdminParams be called twice if we give some pulsar admin config.

Reply:
Use `Option(getAdminParams(parameters)).filter(_.nonEmpty).getOrElse(clientParams)` would be better.

Reply:
Ok, fixed this.